nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPath.java

```diff
@@ -314,10 +331,15 @@ private static Root populateElementsRec(Node node, List<AccessPathElement> eleme
       }
       result = populateElementsRec(accessNode.getReceiver(), elements);
       elements.add(accessPathElement);
+    } else if (node instanceof MethodAccessNode) {
```

Comment:
In method ```onMatchMethodInvocation``` for ```RequiresNonNullHandler```, I use the ```getMethodSelect()``` to get access to the receiver, and it returns an ```ExpressionTree``` which maps to a ```MethodAccessNode``` when calling ```getNodesForTree``` in ```AnalysisResult``` class.
Here are instances in tests that all maps to ```MethodAccessNode```:

1. ```run```
2. ```content.run```
3. ```office2.box.content.run```

I was thinking maybe instead of having a switch case in ```RequiresNonNullHandler``` to processes all types of possible ```ExpressionTree```s we might encounter, it is better to rely on completeness of ```populateElementsRec``` and add this special case to it. Should I follow another approach ?

Reply:
Ah, ok, I think there is a bug here.  `getMethodSelect()` is returning an expression referencing the method itself that is invoked at the call.  Instead, we want the expression referencing the receiver argument of the call.  So, for a call `content.run()`, the `ExpressionTree` we want is for `content`, not `content.run`.  This is because if the `run()` method is annotated `@RequiresNonNull("nullItem")`, we want to check the nullability of `content.nullItem`, not `content.run.nullItem` (which is malformed).  Does that make sense?  For an invocation like `run()` with no explicit receiver argument, we need to figure out that the receiver is `this` and call into the dataflow analysis appropriately.